h2o-core/src/main/java/hex/ParallelModelBuilder.java

```diff
@@ -0,0 +1,56 @@
+package hex;
+
+import jsr166y.ForkJoinTask;
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class ParallelModelBuilder extends ForkJoinTask<ParallelModelBuilder> {
+
+  private BiConsumer<Model, ParallelModelBuilder> _modelFeeder;
+  private Lock _waitingModelsLock;
+
+  public ParallelModelBuilder(BiConsumer<Model, ParallelModelBuilder> modelFeeder) {
+    Objects.requireNonNull(modelFeeder);
+    _modelFeeder = modelFeeder;
+    _waitingModelsLock = new ReentrantLock();
+  }
+
+  public void run(ModelBuilder[] modelBuilders) {
```

Comment:
To make this an isolated showcase, I had to remove few lines of code. This makes the impression the lock is there in vain. In this example, it is less useful then it seems, but it serves one purpose -prevents start of training of multiple models at once, in case two model feeders invoke this method at similar point in time.